improver_tests/utilities/test_cube_extraction.py

```diff
@@ -483,5 +488,128 @@ def test_single_threshold_use_original_units(self):
         )
 
 
+class Test_subset_data(IrisTest):
+    """Tests for the subset_data function"""
+
+    def setUp(self):
+        """Set up spot and gridded cubes for testing"""
+        # details pulled from verification site list
+        alts = np.array([15, 82, 0, 4, 15, 269], dtype=np.float32)
+        lats = np.array([60.75, 60.13, 58.95, 57.37, 58.22, 57.72], dtype=np.float32)
+        lons = np.array([-0.85, -1.18, -2.9, -7.40, -6.32, -4.90], dtype=np.float32)
+        wmo_ids = np.array(["3002", "3005", "3017", "3023", "3026", "3031"])
+
+        self.spot_cube = build_spotdata_cube(
+            np.arange(6).astype(np.float32),
+            "screen_temperature",
+            "degC",
+            alts,
+            lats,
+            lons,
+            wmo_ids,
+        )
+
+        data = np.arange(56).reshape((7, 8)).astype(np.float32)
+        self.uk_cube = set_up_variable_cube(
+            data,
+            name="screen_temperature",
+            units="degC",
+            spatial_grid="equalarea",
+            domain_corner=(-5000, -5000),
+            grid_spacing=2000,
+        )
+
+        self.gl_cube = set_up_variable_cube(
+            data,
+            name="screen_temperature",
+            units="degC",
+            spatial_grid="latlon",
+            domain_corner=(45, -2),
+            grid_spacing=2,
+        )
+
+        self.grid_spec = {
+            "longitude": {"min": 0, "max": 7, "thin": 2},
+            "latitude": {"min": 42, "max": 52, "thin": 2},
+            "projection_x_coordinate": {"min": 0, "max": 10000, "thin": 3},
+            "projection_y_coordinate": {"min": 0, "max": 10000, "thin": 3},
+        }
+
+    def test_subset_spot_cube(self):
+        """Extract a list of spot sites"""
+        site_list = ["3005", "3023"]
+        result = subset_data(self.spot_cube, site_list=site_list)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayEqual(result.coord("wmo_id").points, site_list)
+        self.assertArrayAlmostEqual(result.data, [1, 3])
+
+    def test_subset_uk_grid(self):
+        """Extract subset of UK equal area grid"""
+        expected_data = np.array([[27, 30], [51, 54]])
+        result = subset_data(self.uk_cube, grid_spec=self.grid_spec)
+        self.assertArrayAlmostEqual(result.data, expected_data)
+
+    def test_subset_global_grid(self):
```

Comment:
This test does actually cover the need for `cube.intersection`.  I found that `cube.extract` does not work at all for longitude, and consistently returns `None`.

Reply:
I think that adding a Pacific Ocean unit test (i.e. one that involves crossing the dateline) is important because I think that the usage of `intersection`, as described [in Iris](https://scitools-iris.readthedocs.io/en/stable/generated/api/iris/cube.html?highlight=intersection#iris.cube.Cube.intersection), is most relevant when looking at e.g. a Pacific Ocean domain. Otherwise, the usage of `intersection` can be replaced with `cube.extract`. I've tried this in `_create_cutout` (shown below) and this seems to work:

```python
        lat_constraint = Constraint(latitude=lambda x: ymin <= x.point <= ymax)
        lon_constraint = Constraint(longitude=lambda x: xmin <= x.point <= xmax)
        cutout = cube.extract(lat_constraint & lon_constraint)
```

An example unit test is:
```python
    def test_subset_pacific_global_grid(self):
        """Extract subset of global lat-lon grid"""
        expected_data = np.array([[2., 4.], [18., 20.]])
        grid_spec = {
            "longitude": {"min": 178, "max": 185, "thin": 2},
            "latitude": {"min": 0, "max": 7, "thin": 2},
        }
        result = subset_data(self.gl_pacific_cube, grid_spec=grid_spec)
        self.assertArrayAlmostEqual(result.data, expected_data)
```
Note that the longitude must be specified as something like `{"min": -182, "max": -175, "thin": 2}` i.e. where either the min or max is outside the -180 to 180 range, so as to maintain the minimum being lower than the maximum, otherwise [this Iris exception is raised](https://github.com/SciTools/iris/blob/master/lib/iris/cube.py#L2942). The resulting longitudes are then potentially outside the -180 to 180 range, although are rollable to be from -180 to 180.

Would it be possible to add a unit test, similar to as shown above, to demonstrate this behaviour? Should we put the longitude back into the -180 to 180 range?

Reply:
Lucky you.  Extract on longitudes (as you've written above) didn't work for me, and took a pretty long time to track down!  But yes, if the behaviour isn't reproducible I should certainly add that test.